src/components/image/index.jsx

```diff
@@ -2,20 +2,124 @@ import PropTypes from "prop-types";
 
 const COMPONENT_CLASS_NAME = "c-image";
 
-const Image = ({ alt, className, height, loading, src, width }) => (
-	<img
-		alt={alt}
-		className={className ? `${COMPONENT_CLASS_NAME} ${className}` : `${COMPONENT_CLASS_NAME}`}
-		height={height}
-		loading={loading}
-		src={src}
-		width={width}
-	/>
-);
+const Image = ({
+	alt,
+	className,
+	loading,
+	src,
+	resizedOptions,
+	resizerURL,
+	responsiveImages,
+	width,
+	height,
+	sizes,
+}) => {
+	// get the default height and width of the image
+	// use aspect ratio to generate other sizes
+	const { auth } = resizedOptions;
+
+	if (!auth) {
+		// eslint-disable-next-line no-console
+		console.error("No auth token provided for resizer");
+
+		return (
+			<img
+				alt={alt}
+				className={className ? `${COMPONENT_CLASS_NAME} ${className}` : COMPONENT_CLASS_NAME}
+				src={src}
+				width={width}
+				height={height}
+			/>
+		);
+	}
+
+	// ex: 'filter=filterHere&auth=abc123'
+	const stringOptionsWithoutHeightWidth = new URLSearchParams(resizedOptions).toString();
+
+	// "https://resizer.com" + "\image.jpg" + "?auth=secret&filter=true"
+	const srcWithOptionsWithoutHeightWidth = resizerURL.concat(
+		src,
+		"?",
+		stringOptionsWithoutHeightWidth
+	);
+
+	let aspectRatio = 0;
+
+	// if no height and width, no responsive image calculation
+	if (height && width) {
+		aspectRatio = width / height;
+	}
+
+	const responsiveHeightsAndWidths = responsiveImages.reduce(
+		(accumulator, responsiveImageWidth) => {
+			if (Number.isInteger(responsiveImageWidth) && responsiveImageWidth > 0) {
+				return [
+					...accumulator,
+					{
+						width: responsiveImageWidth,
+						// divide the derived aspect ratio by each of the responsiveImages widths to get the height
+						// aspect ratio of zero will not show the height
+						...(aspectRatio !== 0 && { height: responsiveImageWidth / aspectRatio }),
+					},
+				];
+			}
+			return accumulator;
+		},
+		[]
+	);
+
+	// add the height and width to the default src if they exist
+	// auth token will at least exist here so don't need to worry about ? prepend
+	const defaultSrc = srcWithOptionsWithoutHeightWidth.concat(
+		`${width ? `&width=${width}` : ""}${height ? `&height=${height}` : ""}`
+	);
+
+	return (
+		<img
+			alt={alt}
+			className={className ? `${COMPONENT_CLASS_NAME} ${className}` : COMPONENT_CLASS_NAME}
+			height={height}
+			loading={loading}
+			src={defaultSrc}
+			width={width}
+			srcSet={
+				responsiveHeightsAndWidths
```

Comment:
conflicting prettier and eslint tab controls. and I found this eslint-prettier-config cli tool to resolve the diffs